File: services/payment/src/main/java/com/example/iter/payment/dto/toss/TossConfirmApiResponse.java
package com.example.iter.payment.dto.toss;

import java.time.LocalDateTime;
import java.time.OffsetDateTime;

public record TossConfirmApiResponse(String paymentKey, String orderId, String status, String approvedAt, Long totalAmount) {
    public LocalDateTime approvedAtAsLocalDateTime() {
        return OffsetDateTime.parse(approvedAt).toLocalDateTime();
    }
}

File: services/payment/src/main/kotlin/com/example/iter/payment/dto/toss/TossConfirmApiResponse.kt
package com.example.iter.payment.dto.toss

import java.time.LocalDateTime
import java.time.OffsetDateTime

@JvmRecord
data class TossConfirmApiResponse(
    val paymentKey: String,
    val orderId: String,
    val status: String,
    val approvedAt: String?,
    val totalAmount: Long,
) {
    fun approvedAtAsLocalDateTime(): LocalDateTime = OffsetDateTime.parse(approvedAt).toLocalDateTime()
}
